test/core/2_compound-market.spec.ts

```diff
@@ -318,4 +318,58 @@ describe('compound-market', async () => {
       router.createMarket(consts.MARKET_FACTORY_COMPOUND, xyt.address, testToken.address, consts.HG)
     ).to.be.revertedWith('EXISTED_MARKET');
   });
+
+  it("shouldn't be able to create markets if tokens, factoryId are invalid", async() => {
+    await expect(router.createMarket(consts.MARKET_FACTORY_COMPOUND, consts.ZERO_ADDRESS, testToken.address, consts.HG)).to.be.revertedWith("ZERO_ADDRESS");
+    await expect(router.createMarket(consts.MARKET_FACTORY_COMPOUND, xyt.address, consts.ZERO_ADDRESS, consts.HG)).to.be.revertedWith("ZERO_ADDRESS");
+    await expect(
+      router.createMarket(consts.MARKET_FACTORY_COMPOUND, consts.RANDOM_ADDRESS, testToken.address, consts.HG)
+    ).to.be.revertedWith('INVALID_YT');
+    await expect(
+      router.createMarket(consts.ZERO_BYTES, xyt.address, testToken.address, consts.HG)
+    ).to.be.revertedWith('ZERO_ADDRESS');
+    await expect(
+      router.createMarket(consts.MARKET_FACTORY_AAVE_V2, xyt.address, testToken.address, consts.HG)
+    ).to.be.revertedWith('INVALID_FORGE_FACTORY');
+  })
+
+  it('router should reject dual liquidity addition if token amount is invalid or market does not exist', async() => {
+    await expect(router.addMarketLiquidityDual(consts.MARKET_FACTORY_COMPOUND, xyt.address, testToken.address, BN.from(100), BN.from(100), consts.INF, BN.from(0))).to.be.revertedWith("INVALID_YT_AMOUNTS");
+    await expect(router.addMarketLiquidityDual(consts.MARKET_FACTORY_COMPOUND, xyt.address, testToken.address, BN.from(100), BN.from(0), BN.from(0), BN.from(0))).to.be.revertedWith("INVALID_TOKEN_AMOUNTS");
+    await expect(router.addMarketLiquidityDual(consts.MARKET_FACTORY_COMPOUND, xyt.address, consts.RANDOM_ADDRESS, BN.from(100), consts.INF, BN.from(100), BN.from(0))).to.be.revertedWith("MARKET_NOT_FOUND");
+  })
```

Comment:
the swap function in contract functions should also work here. Do change other checks as well.